csrc/scheduler/reduction_heuristic.h

```diff
@@ -310,7 +319,8 @@ class ReductionParams : public HeuristicParams {
   }
 
   std::shared_ptr<HeuristicParams> clone() const override {
-    return std::make_shared<ReductionParams>(*this);
+    auto ptr = std::make_shared<ReductionParams>(*this);
+    return std::static_pointer_cast<HeuristicParams>(ptr);
```

Comment:
out of curiosity, is there a linter or a compiler that's complaining on this?

Reply:
yes, clangtidy